src/js/core/DCOSApplication.js

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable no-unused-vars */
+import React from "react";
+/* eslint-enable no-unused-vars */
+import ReactDOM from "react-dom";
+import { decorate, injectable, inject } from "inversify";
+import { IntlProvider } from "react-intl";
+import { RequestUtil } from "mesosphere-shared-reactjs";
+import { Router, hashHistory } from "react-router";
+import { Provider } from "react-redux";
+import PluginSDK from "PluginSDK";
+
+// Load in our CSS.
+// TODO - DCOS-6452 - remove component @imports from index.less and
+// require them in the component.js
+import "../../styles/index.less";
+import "../utils/MomentJSConfig";
+import { CONFIG_ERROR } from "../constants/EventTypes";
+import ApplicationUtil from "../utils/ApplicationUtil";
+import appRoutes from "../routes/index";
+import ConfigStore from "../stores/ConfigStore";
+import RequestErrorMsg from "../components/RequestErrorMsg";
+
+// Translations
+import enUS from "../translations/en-US.json";
+
+const domElement = global.document.getElementById("application");
+const navigatorLanguage = "en-US";
+
+// TODO: Implement loader that can concat many sprites into a single one
+// We opt to load the sprite after the Javscript files are parsed because it
+// is quite expensive for the browser to parse a sprite file. This way we
+// don't block the JS execution.
+setTimeout(function() {
+  var ajax = new XMLHttpRequest();
+  ajax.open("GET", "sprite.svg", true);
+  ajax.send();
+  ajax.onload = function() {
+    var div = global.document.createElement("div");
+    div.innerHTML = ajax.responseText;
+    div.style.height = 0;
+    div.style.overflow = "hidden";
+    div.style.width = 0;
+    div.style.visibility = "hidden";
+    global.document.body.insertBefore(div, global.document.body.childNodes[0]);
+  };
+});
+
+// Patch json
+const oldJSON = RequestUtil.json;
+RequestUtil.json = function(options = {}) {
+  // Proxy error function so that we can trigger a plugin event
+  const oldHandler = options.error;
+  options.error = function() {
+    if (typeof oldHandler === "function") {
+      oldHandler.apply(null, arguments);
+    }
+    PluginSDK.Hooks.doAction("AJAXRequestError", ...arguments);
+  };
+
+  oldJSON(options);
+};
+
+export default class DCOSApplication {
+  constructor(routingService, navigationService) {
+    this._routingService = routingService;
+    this._navigationService = navigationService;
+  }
+
+  onStart() {
+    if (!global.Intl) {
+      require.ensure(["intl", "intl/locale-data/jsonp/en.js"], function(
+        require
+      ) {
+        require("intl");
+        require("intl/locale-data/jsonp/en.js");
+
+        this.startApplication();
+      });
+    } else {
+      this.startApplication();
+    }
+  }
+
+  renderAppToDOM(content) {
```

Comment:
This name is ambitious, I would not know how to distinguish `renderAppToDom` and `renderApplicationToDom`

Reply:
I didn't mean refactoring the logic, I just wrapped it with a class.

Reply:
Makes sense 👍